envoy/README.md

```diff
@@ -0,0 +1,520 @@
+<div align="center">
+<h1>PATH<br/>Authorization & Rate Limiting</h1>
+<img src="https://storage.googleapis.com/grove-brand-assets/Presskit/Logo%20Joined-2.png" alt="Grove logo" width="500"/>
+</div>
+<br/>
+
+# Table of Contents <!-- omit in toc -->
+
+- [1. Overview](#1-overview)
+  - [1.1. Components](#11-components)
+  - [1.2 URL Format](#12-url-format)
+- [2. Quickstart](#2-quickstart)
+- [3. Envoy Proxy](#3-envoy-proxy)
+  - [3.1. Contents](#31-contents)
+  - [3.2. Envoy HTTP Filters](#32-envoy-http-filters)
+  - [3.3. Request Lifecycle](#33-request-lifecycle)
+- [4. Gateway Endpoint Authorization](#4-gateway-endpoint-authorization)
+  - [4.1 JSON Web Token (JWT) Authorization](#41-json-web-token-jwt-authorization)
+  - [4.2 API Key Authorization](#42-api-key-authorization)
+  - [4.3 No Authorization](#43-no-authorization)
+- [5. External Authorization Server](#5-external-authorization-server)
+  - [5.1. Gateway Endpoints gRPC Service](#51-gateway-endpoints-grpc-service)
+  - [5.2. Remote gRPC Auth Server](#52-remote-grpc-auth-server)
+    - [5.2.1. PATH Auth Data Server](#521-path-auth-data-server)
+    - [5.2.2. Example Gateway Endpoint YAML File](#522-example-gateway-endpoint-yaml-file)
+    - [5.2.3. Custom Remote gRPC Server Implementation](#523-custom-remote-grpc-server-implementation)
+- [6. Rate Limiter](#6-rate-limiter)
+  - [6.1. Rate Limit Configuration](#61-rate-limit-configuration)
+  - [6.2. Documentation and Examples](#62-documentation-and-examples)
+
+## 1. Overview
+
+This folder contains everything necessary for managing authorization and rate limiting in the PATH service.
+Specifically, this is split into two logical parts:
+
+1. The `Envoy Proxy configuration`
+2. The `Go External Authorization Server`
+
+### 1.1. Components
+
+- **PATH Service**: The service that handles requests after they have been authorized.
+- **Envoy Proxy**: A proxy server that handles incoming requests, performs auth checks, and routes authorized requests to the `PATH` service.
+- **External Authorization Server**: A Go/gRPC server that evaluates whether incoming requests are authorized to access the `PATH` service.
+- **Rate Limiter**: A service that coordinates all rate limiting.
+- **Redis**: A key-value store used by the rate limiter to share state and coordinate rate limiting among all services.
+- **Remote gRPC Server** _(Implemented by Gateway Operator)_: A server that provides the external authorization server with data on which endpoints are authorized to use the PATH service.
```

Comment:
Wdyt of updating this part and simply mentioning and linking to PADS?

Reply:
Makes sense. This terminology predates PADS so I will update the readme.